lib/metrics/index.js

```diff
@@ -0,0 +1,144 @@
+'use strict';
+
+const PrometheusClient = require('./clients/prometheus');
+const StatsDClient = require('./clients/statsd');
+const LogClient = require('./clients/log');
+const Metric = require('./metric');
+const normalizeName = require('./utils').normalizeName;
+
+const SUPPORTED_CLIENTS = {
+    PROMETHEUS: 'prometheus',
+    STATSD: 'statsd',
+    LOG: 'log'
+};
+
+const DEPRECATED_METHODS = ['timing', 'increment', 'decrement', 'gauge', 'unique'];
+
+class Metrics {
+    constructor(options, logger) {
+        this.logger = logger;
+        this.cache = new Map();
+        this.clients = [];
+        options.forEach((o) => {
+            if (o.type === SUPPORTED_CLIENTS.PROMETHEUS) {
+                this.clients.push(new PrometheusClient(o, this.logger));
+            }
+            if (o.type === SUPPORTED_CLIENTS.STATSD) {
+                this.clients.push(new StatsDClient(o, this.logger));
+            }
+            if (o.type === SUPPORTED_CLIENTS.LOG) {
+                o.methods = DEPRECATED_METHODS;
+                this.clients.push(new LogClient(o, this.logger));
+            }
+        });
+    }
+
+    fetchClient(name) {
+        return this.clients.find((client) => {
+            return client.constructor.name === name;
+        });
+    }
```

Comment:
Why do we need to expose a 'client'? It seems to be an internal thingy.

Reply:
Our ultimate goal is to eliminate StatsD entirely.  The idea here was to allow users the option to get to the raw client if needed rather than hide the abstraction completely and leave users no other option than to implement it themselves.

Reply:
I think it's reasonable to assume that (i) users want as simplistic as an interface as possible; and (ii) prometheus will not be available in all environments (dev, local, vagrant, etc). In light of that, I would opt for hiding the prometheus logic and client away and provide simple ways for people to send metrics.